datahub/opal.py

```diff
@@ -90,13 +97,18 @@ def get_opal_row(data: dict[str, float]) -> pd.Series:  # type: ignore[type-arg]
     Returns:
         A pandas Series containing the new data
     """
-    data_array = []
+    if isinstance(data, dict):
+        data_index = data["frame"]
+        data_array = [data[item] for item in opal_headers.values()]
+
+    else:
+        data_array = data
```

Comment:
Just need to be careful here. Because you delete elements from `data_array` later on, they are also deleted from `data` unless you explicitly make a copy of it like this.

```suggestion
        data_array = data.copy()
```